### PR TITLE
[6X] fix incorrect scan position during bitmap index words scan

### PR DESCRIPTION
**Pull request by SmartKeyerror**
This is the backport of #13479.

Since this PR https://github.com/greenplum-db/gpdb/pull/11377 has fixed bitmap index scan when concurrent insert update full
bitmap page, which resolved concurrent read on bitmap index when there's insert running
in the backend may cause the bitmap scan read wrong tid:

1. Query on bitmap: A query starts and reads all bitmap pages to `PAGE_FULL`, increases
the next tid to fetch, and releases the lock after reading each page.
2. Concurrent insert: insert a tid into `PAGE_FULL` cause expand compressed words to
new words, and rearrange words into `PAGE_NEXT`.
3. Query on bitmap: fetch `PAGE_NEXT` and expect the first tid in it should equal the
saved next tid. But actually `PAGE_NEXT` now contains words used to belong in `PAGE_FULL`.
This causes the real next tid less than the expected next tid. But our scan keeps increasing
the wrong tid. And then this leads to a wrong result.

The PR used `_bitmap_catchup_to_next_tid()` function to adjust `result->lastScanWordNo`
to the correct position if there has concurrent read/write and causes rearranging words into the
next bitmap index page, it used the value of `words->firstTid` and `result->nextTid` to judge
whether rearranging words into the next bitmap index page happened or not, this is not entirely
right.

This is because `BMIterateResult` can only store 16*1024=16384 TIDs, but `BMBatchWords`
can save 3968 words by default, a words is 64bit, even in the worst case (all words not
compressed) , `BMBatchWords` can hold 3968 * 64 = 253952 TIDs. So if there has a `PAGE_FULL`,
the `PAGE_FULL` must scan it more than one time, but the value of `BMBatchWords->firstTid` will
not be updated during each scan, it will only be updated when new bitmap index pages are read.

Therefore, in the absence of concurrent read/write, if we need to scan the same `BMBatchWords`
multiple times, it will lead to the wrong scan position, resulting in wrong output results, as https://github.com/greenplum-db/gpdb/issues/13446.

To summarize, we just need to check for a rearranged condition when the new bitmap index page is
read from disk, we should do nothing when we scan the same `BMBatchWords`.

### CODE REVIEW

**Review by interma — APPROVED** [review]
LGTM

It's the same as the master PR.

**Comment by SmartKeyerror** [comment]
My local full pipeline is all green, so merge it.
